weather_mv/loader_pipeline/regrid.py

```diff
@@ -78,34 +226,49 @@ def apply(self, uri: str):
         if self.dry_run:
             return
 
-        logger.debug(f'Copying grib from {uri!r} to local disk.')
-        with open_local(uri) as local_grib:
-            # TODO(alxr): Figure out way to open fieldset in memory...
-            logger.debug(f'Regridding {uri!r}.')
-            try:
+        with _metview_op():
+            logger.debug(f'Copying grib from {uri!r} to local disk.')
+
+            with open_local(uri) as local_grib:
+                # TODO(alxr): Figure out way to open fieldset in memory...
+                logger.debug(f'Regridding {uri!r}.')
                 fs = mv.bindings.Fieldset(path=local_grib)
                 fieldset = mv.regrid(data=fs, **self.regrid_kwargs)
-            except (ModuleNotFoundError, ImportError, FileNotFoundError) as e:
-                raise ImportError('Please install MetView with Anaconda:\n'
-                                  '`conda install metview-batch -c conda-forge`') from e
 
-        with tempfile.NamedTemporaryFile() as src:
-            logger.debug(f'Writing {self.target_from(uri)!r} to local disk.')
-            if self.to_netcdf:
-                fieldset.to_dataset().to_netcdf(src.name)
-            else:
-                mv.write(src.name, fieldset)
+            with tempfile.NamedTemporaryFile() as src:
+                logger.debug(f'Writing {self.target_from(uri)!r} to local disk.')
+                if self.to_netcdf:
+                    fieldset.to_dataset().to_netcdf(src.name)
+                else:
+                    mv.write(src.name, fieldset)
 
-            src.flush()
+                src.flush()
 
-            # Clear the metview temporary directory.
-            cache_dir = glob.glob(f'{tempfile.gettempdir()}/mv*')[0]
-            shutil.rmtree(cache_dir)
-            os.makedirs(cache_dir)
+                _clear_metview()
 
-            logger.info(f'Uploading {self.target_from(uri)!r}.')
-            with FileSystems().create(self.target_from(uri)) as dst:
-                shutil.copyfileobj(src, dst, WRITE_CHUNK_SIZE)
+                logger.info(f'Uploading {self.target_from(uri)!r}.')
+                with FileSystems().create(self.target_from(uri)) as dst:
+                    shutil.copyfileobj(src, dst, WRITE_CHUNK_SIZE)
 
     def expand(self, paths):
-        paths | beam.Map(self.apply)
+        if not self.zarr:
+            paths | beam.Map(self.apply)
+            return
+
+        source_ds = xr.open_zarr(self.first_uri, **self.zarr_kwargs, chunks=None)
```

Comment:
You're opening a file in expand? That seems less than ideal...

Reply:
Sort of. Mostly, I'm getting the Zarr metadata, which is necessary to open the dataset to chunks. 

https://docs.xarray.dev/en/stable/generated/xarray.open_zarr.html#:~:text=If%20chunks%3DNone%2C%20zarr%20array%20data%20will%20lazily%20convert%20to%20numpy%20arrays%20upon%20access

I'll add a comment here to clarify what's going on.